app/src/main/java/org/oppia/android/app/devoptions/markchapterscompleted/MarkChaptersCompletedFragmentPresenter.kt

```diff
@@ -94,14 +95,14 @@ class MarkChaptersCompletedFragmentPresenter @Inject constructor(
   }
 
   private fun createRecyclerViewAdapter(): BindableAdapter<MarkChaptersCompletedItemViewModel> {
-    return BindableAdapter.MultiTypeBuilder
-      .newBuilder<MarkChaptersCompletedItemViewModel, ViewType> { viewModel ->
-        when (viewModel) {
-          is StorySummaryViewModel -> ViewType.VIEW_TYPE_STORY
-          is ChapterSummaryViewModel -> ViewType.VIEW_TYPE_CHAPTER
-          else -> throw IllegalArgumentException("Encountered unexpected view model: $viewModel")
-        }
+    return multiTypeBuilderFactory.create<MarkChaptersCompletedItemViewModel,
```

Comment:
Ditto here--let's see if we can wrap this a bit more cleanly.

Reply:
Changes applied.